pkg/apiserver/resthandler.go

```diff
@@ -488,8 +488,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		gv := scope.Kind.GroupVersion()
 		codec := runtime.NewCodec(
 			scope.Serializer.EncoderForVersion(s, gv),
-			scope.Serializer.DecoderToVersion(s, unversioned.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
-		)
+			scope.Serializer.DecoderToVersion(s, gv))
```

Comment:
#25106 is likely something client-side in apply or validate, not in the resthandler. I think this PR is exercising the wrong area